mne/annotations.py

```diff
@@ -0,0 +1,49 @@
+# Authors: Jaakko Leppakangas <jaeilepp@student.jyu.fi>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from datetime import datetime
+import time
+
+
+class Annotations(object):
+    """Annotation object for annotating segments of raw data.
+
+    Parameters
+    ----------
+    onset: array of float
+        Annotation time onsets from the beginning of the recording.
+    duration: array of float
+        Durations of the annotations.
+    description: array of str
+        Array of strings containing description for each annotation.
+    orig_time: int | instance of datetime | None
+        Timestamp or a datetime determining the starting time of annotation
+        acquisition. If None (default), starting time is determined from
+        beginning of raw data.
+    """
+
+    def __init__(self, onset, duration, description, orig_time=None):
+
+        if orig_time is not None:
+            if isinstance(orig_time, datetime):
+                orig_time = time.mktime(orig_time.timetuple())
+            elif not np.isscalar(orig_time):
```

Comment:
I personally like the idea of constructing an annotation object and then be able to use it straight away without having to do any adjustments. There is also the mess with first_samps. I think the annotation files will usually have time stamps, so this should work fine.
But yeah, there should be more accurate time shifts here. At least millisecond accuracy.